primitives/ed25519/ed25519.go

```diff
@@ -485,6 +486,37 @@ func (priv PrivateKey) Sign(rand io.Reader, message []byte, opts crypto.SignerOp
 // PublicKey is the type of Ed25519 public keys.
 type PublicKey []byte
 
+// String returns a string representation of the public key.
+func (k PublicKey) String() string {
+	return base64.StdEncoding.EncodeToString(k)
+}
+
+// MarshalText encodes a public key into text form.
+func (k PublicKey) MarshalText() (data []byte, err error) {
+	return []byte(base64.StdEncoding.EncodeToString(k)), nil
+}
+
+// UnmarshalText decodes a text marshaled public key.
+func (k *PublicKey) UnmarshalText(text []byte) error {
+	b, err := base64.StdEncoding.DecodeString(string(text))
+	if err != nil {
+		return err
+	}
+
+	return k.UnmarshalBinary(b)
+}
+
+// UnmarshalBinary decodes a binary marshaled public key.
+func (k *PublicKey) UnmarshalBinary(data []byte) error {
+	if len(data) != PublicKeySize {
+		return fmt.Errorf("malformed public key")
+	}
+
+	copy(*k, data)
```

Comment:
If you wanted, you could do additional verification here and reject blatantly malformed public keys.

The relevant helper is:
https://github.com/oasisprotocol/curve25519-voi/blob/1f23a7beb09a352354cbd7277c6b7f0170cf6d5d/primitives/ed25519/ed25519.go#L557

Allowing small order/non-canonical A at the s11n stage is "fine" (though such things will never be produced by correct implementations), because the check also happens at verification time.